client/pages/add-exercises.jsx

```diff
@@ -5,33 +5,108 @@ export default class AddExercisePage extends React.Component {
     super(props);
     this.state = {
       exerciseName: '',
-      exercises: []
+      exercises: [],
+      data: {
+        exercises: []
+      }
     };
     this.handleExerciseNameChange = this.handleExerciseNameChange.bind(this);
     this.handleSubmit = this.handleSubmit.bind(this);
     this.handleClick = this.handleClick.bind(this);
+    this.handleRepsChange = this.handleRepsChange.bind(this);
+    this.handleWeightChange = this.handleWeightChange.bind(this);
+    this.handleIconChange = this.handleIconChange.bind(this);
   }
 
   handleExerciseNameChange(event) {
     this.setState({ exerciseName: event.target.value });
   }
 
+  handleRepsChange(index, event) {
+    const dataCopy = Object.assign({}, this.state.data);
+    dataCopy.exercises[index].reps = event.target.value;
+    this.setState({ data: dataCopy });
+  }
+
+  handleWeightChange(index, event) {
+    const dataCopy = Object.assign({}, this.state.data);
+    dataCopy.exercises[index].weight = event.target.value;
+    this.setState({ data: dataCopy });
+  }
+
+  handleIconChange(index, event) {
```

Comment:
This method doesn't say what is really changing. it's not the icon, it's the completed status of the set